nightfall-optimist/src/event-handlers/rollback.mjs

```diff
@@ -36,131 +33,52 @@ async function rollbackEventHandler(data) {
-      maybeValidTransactions.push(t);
-    }
-  });
 
+  const invalidTransactions = [];
   // For valid transactions that have made it to this point, we run them through our transaction checker for validity
-  for (let i = 0; i < maybeValidTransactions.length; i++) {
-    let checkTransactionCorrect = true;
+  for (let i = 0; i < blocksToBeDeleted.length; i++) {
+    // Get the trannsaction hashes included in these blocks
+    const transactionHashesInBlock = blocksToBeDeleted[i].transactionHashes.flat(Infinity);
+    // Use the transaction hashes to grab the actual transactions filtering out deposits - In Order.
+    // eslint-disable-next-line no-await-in-loop
+    const blockTransactions = (await getTransactionsByTransactionHashes(transactionHashesInBlock)) // TODO move this to getTransactionsByTransactionHashes by l2 block number because transaction hash is not unique and might not pull the right l2 block number
+      .filter(t => t.transactionType !== '0');
+    logger.info(`blockTransctions: ${JSON.stringify(blockTransactions)}`);
+    for (let j = 0; j < blockTransactions.length; j++) {
+      try {
+        // eslint-disable-next-line no-await-in-loop
+        await checkTransaction(blockTransactions[j], false, {
+          blockNumberL2: blocksToBeDeleted[i].blockNumberL2,
+        });
+      } catch (error) {
+        logger.debug(`Invalid checkTransaction: ${blockTransactions[j].transactionHash}`);
+        invalidTransactions.push(blockTransactions[j].transactionHash);
+      }
+    }
     try {
-      // eslint-disable-next-line no-await-in-loop
-      await checkTransaction(maybeValidTransactions[i]);
+      checkDuplicateCommitmentsWithinBlock(blocksToBeDeleted[i], blockTransactions);
+      checkDuplicateNullifiersWithinBlock(blocksToBeDeleted[i], blockTransactions);
     } catch (error) {
-      checkTransactionCorrect = false;
-    }
-    if (!checkTransactionCorrect) {
-      logger.debug(`Invalid checkTransaction: ${maybeValidTransactions[i].transactionHash}`);
-      invalidTransactionSet.add(maybeValidTransactions[i].transactionHash);
+      const { transaction2: transaction } = error.metadata; // TODO pick transaction to delete based on which transaction pays more to proposer
+      logger.debug(`Invalid checkTransaction: ${transaction.transactionHash}`);
+      invalidTransactions.push(transaction.transactionHash);
     }
   }
-  const invalidTransactionHashesArr = [...invalidTransactionSet];
-  logger.debug(`Deleting transactions: ${invalidTransactionHashesArr}`);
-  await deleteTransactionsByTransactionHashes(invalidTransactionHashesArr);
 
-  logger.debug(
-    `Nullifier of deleted transactions: ${
-      transactions.find(t => t.transactionHash === invalidTransactionHashesArr[0])?.nullifiers[0]
-    }`,
+  // We can now reset or delete the local database.
+  await Promise.all(
+    blocksToBeDeleted
+      .map(async block => [addTransactionsToMemPool(block), deleteBlock(block.blockHash)])
+      .flat(1),
```

Comment:
`1` is a default value for `flat()` so could easily be skipped.

Reply:
True. It is a minor change, so I don't want to kick start the github action tests again because it has been waiting to be merged for 1.5 months now